rmf_task/test/integration/test_backups.cpp

```diff
@@ -15,24 +15,64 @@
  *
 */
 
+#include <iostream>
+#include <filesystem>
 #include <rmf_task/BackupFileManager.hpp>
 
 #include "../mock/MockDelivery.hpp"
 
 #include <rmf_utils/catch.hpp>
 
+const std::string backup_root_dir = "/tmp/rmf_task/test_backups";
+void cleanup() {
+  std::filesystem::remove_all(std::filesystem::weakly_canonical(backup_root_dir));
+}
+
+SCENARIO("Folder creation tests")
+{
+  cleanup();
+
+  rmf_task::BackupFileManager backup(backup_root_dir);
+  backup.clear_on_startup(true);
+  backup.clear_on_shutdown(true);
+
+  auto group_backup = backup.make_group("group");
+  CHECK(std::filesystem::exists(backup_root_dir + "/group"));
+
+  CHECK(!std::filesystem::exists(backup_root_dir + "/group" + "/robot"));
+  group_backup->make_robot("robot"); // Due to RAII and clear_on_shutdown the destructor is called immediately
+  CHECK(!std::filesystem::exists(backup_root_dir + "/group" + "/robot"));
+
+  auto robot_backup = group_backup->make_robot("robot"); 
+  CHECK(std::filesystem::exists(backup_root_dir + "/group" + "/robot"));
```

Comment:
You could add a test that checks that the path is removed after `robot_backup` is cleared, e.g.
```
robot_backup.reset();
CHECK_FALSE(std::filesystem::exists(backup_root_dir / "group" / "robot"));
```

Reply:
https://github.com/open-rmf/rmf_task/commit/6298276399fb291bf20dd1dc59f1cdff3063771d and  0b66d33b04dbeba3cd0215ec1a5ab38f3e104d67 adds this

Reply:
I just ran the test in 6298276399fb291bf20dd1dc59f1cdff3063771d and I see that the robot directory was cleared but not the group directory. That's because the `Robot::Implementation` class has a custom destructor for clearing out the directory while `Group::Implementation` does not.

I think it's okay to not delete group directories, in case anyone wants to store other things in the directory that the `BackupFileManager` isn't involved with. In fact, it might be better to also not delete the robot directory. Instead we can make it so only the `backup` and `.backup` files get deleted.